enterprise/server/api/api_server.go

```diff
@@ -492,6 +492,11 @@ func (s *APIServer) ExecuteWorkflow(ctx context.Context, req *apipb.ExecuteWorkf
 
 	wf, err := wfs.GetWorkflowByGroupAndRepo(ctx, user.GetGroupID(), req.GetRepoUrl())
 	if err != nil {
+		if status.IsNotFoundError(err) {
+			return nil, status.NotFoundErrorf("Workflow for repo %s not found. Note that Github App authentication is required"+
+				" to use this API. If you are using the legacy Workflow product, you will need"+
+				" to migrate to a Github App first.", req.GetRepoUrl())
```

Comment:
The workflows page encourages people to migrate, so we can link people there as an actionable step, and also avoid having to update this messaging later

```suggestion
			return nil, status.NotFoundErrorf("Workflow for repo %q not found. Check the Workflows page in the BuildBuddy app and ensure this repo is properly linked.", req.GetRepoUrl())
```